gammapy/cube/fit.py

```diff
@@ -1418,20 +1418,23 @@ def update(self, exposure, psf, edisp, geom):
         log.debug("Updating model evaluator")
         # cache current position of the model component
 
-        # TODO: lookup correct Edisp for this component
         if isinstance(edisp, EDispMap):
             e_reco = geom.get_axis_by_name("energy").edges
             self.edisp = edisp.get_energy_dispersion(self.model.position, e_reco=e_reco)
         else:
             self.edisp = edisp
 
-        # TODO: lookup correct PSF for this component
-        self.psf = psf  # .get_psf_kernel(self.model.position, geom=exposure.geom)
+        if isinstance(psf, PSFMap):
+            self.psf = psf.get_psf_kernel(
+                self.model.position, geom=exposure.geom, max_radius=0.8 * u.deg
```

Comment:
I would propose to modify `.get_psf_kernel()` and just take the maximum of `table_psf.rad` as a default maximum radius (`if max_radius is  None`). This is more consistent then having a separate hard-coded value here...